@here/harp-mapview/lib/MapView.ts

```diff
@@ -1752,6 +1757,39 @@ export class MapView extends THREE.EventDispatcher {
         return this.m_fog;
     }
 
+    /**
+     * Sets the DataSource which contains the elevations, the elevation range source, and the
+     * elevation provider. Only a single elevation source is possible per [[MapView]]
+     *
+     * If the terrain-datasource is merged with this repository, we could internally construct
+     * the [[ElevationRangeSource]] and the [[ElevationProvider]] and access would be granted to
+     * the application when it asks for it, to simplify the API.
+     *
+     * @param elevationSource The datasource containing the terrain tiles.
+     * @param elevationRangeSource Allows access to the elevation min / max per tile.
+     * @param elevationProvider Allows access to the elevation at a given location or a ray
+     *      from the camera.
+     */
+    setElevationSource(
+        elevationSource: DataSource,
```

Comment:
1. I don't understand why you need three instances of objects here.

2. Looks like they are so interconnected internally, that they should be accessed through unified interface.

3. Isn't `elevationSource` just an implementation detail of how latter (`elevationProvider` and `elevationRangeSource`).

4. Naming. It's really hard to grasp difference between `provider` , `source`.

5. No way to unset these.

Reply:
1. The three instances are doing separate things
- elevationSource is like a DataSource, it has the same interface and shows the mesh rendering
- elevationRangeSource is similar to a DataSource, however it is special, because it controls which requests we can make to the elevationSource (it saves bandwidth to download the bounding box first before we download the tile).
- elevationProvider accesses the heights based on the visible tiles.

Where is your specific misunderstanding?

2. A unified interface sounds good, but there is a high chance of misusing it. Specifically because some parts are async, and some parts are sync. i.e. the getHeight function is meant to be sync, i.e. it can be used to get the value immediately. Splitting like this makes the responsibilities / functionality clear IMO, unless there is some specific example which shows otherwise?

3. elevationSource is the source of data for the terrain tiles, i.e. the meshes

4. appreciate this, if you have a better idea, let me know!!

5. easy to add later if there is a use case which requires this feature.